http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/internal/NettyConfigurationTest.java

```diff
@@ -0,0 +1,107 @@
+package software.amazon.awssdk.http.nio.netty.internal;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static software.amazon.awssdk.http.SdkHttpConfigurationOption.CONNECTION_TIMEOUT;
+import static software.amazon.awssdk.http.SdkHttpConfigurationOption.MAX_CONNECTIONS;
+
+import java.time.Duration;
+import java.util.function.Consumer;
+import org.junit.Test;
+import software.amazon.awssdk.http.nio.netty.NettySdkHttpClientFactory;
+import software.amazon.awssdk.utils.AttributeMap;
+
+public class NettyConfigurationTest {
+
+    @Test
+    public void connectionAcquistion_ReturnsMillis() {
+        NettyConfiguration configuration = createConfiguration(b -> b.connectionAcquisitionTimeout(Duration.ofSeconds(1)));
+        assertEquals(1000, configuration.connectionAcquisitionTimeout());
+    }
+
+    @Test
+    public void connectionAcquisition_PerformsSaturatedCast() {
+        NettyConfiguration configuration = createConfiguration(b -> b.connectionAcquisitionTimeout(Duration.ofMillis(Long.MAX_VALUE)));
+        assertEquals(Integer.MAX_VALUE, configuration.connectionAcquisitionTimeout());
+    }
+
+    @Test
+    public void connectionTimeout_OnlyPulledFromAttributeMap() {
+        NettyConfiguration configuration = createConfiguration(a -> a.put(CONNECTION_TIMEOUT,
+                                                                          Duration.ofSeconds(42)),
+                                                               b ->
+                                                                   b.connectionTimeout(Duration.ofSeconds(9001)));
+        assertEquals(42 * 1000, configuration.connectionTimeout());
+    }
+
+    @Test
+    public void connectionTimeout_PerformsSaturatedCast() {
+        NettyConfiguration configuration = createConfiguration(a -> a.put(CONNECTION_TIMEOUT,
+                                                                          Duration.ofMillis(Long.MAX_VALUE)),
+                                                               b -> {
+                                                               });
+        assertEquals(Integer.MAX_VALUE, configuration.connectionTimeout());
+    }
+
+    @Test
+    public void maxConnectionsPerEndpoint_OnlyPulledFromAttributeMap() {
+        NettyConfiguration configuration = createConfiguration(a -> a.put(MAX_CONNECTIONS, 10),
+                                                               b -> b.maxConnectionsPerEndpoint(11));
+        assertEquals(10, configuration.maxConnectionsPerEndpoint());
+    }
+
+    @Test
+    public void readTimeout_AppliesDefaultValueIfNotSetOnFactory() {
+        NettyConfiguration configuration = createEmptyConfiguration();
+        assertEquals(60, configuration.readTimeout());
```

Comment:
Can we use `AssertJ` matchers here?

That was our original decision at the outset - I know there's been resistance to it and maybe we should have another conversation - but it was always between `AssertJ` and `Hamcrest` we never even talked about using the core `JUnit` asserts (which I personally thing leave a lot to be desired!)

Reply:
vote for `AssertJ`! love the syntax and method chaining. Found an interesting article comparing those two frameworks: https://dzone.com/articles/hamcrest-vs-assertj-assertion-frameworks-which-one

Reply:
Nice write-up - I don't actually really care either way. I can see that AssertJ is a little bit more discoverable but hamcrest is probably (anecdotal) more common (it's been around longer I suppose).

My biggest concern is consistency - we need to make a stand so that when people are contributing they know which framework to use.

Similarly - we should converge on a mocking framework - this one I feel more strongly about - mockito > easymock.

Reply:
+1 haven't done V2 in a while and forgot about assertJ